src/androidTest/java/de/dennisguse/opentracks/content/data/TestDataUtil.java

```diff
@@ -73,4 +79,26 @@ public static void insertTrackWithLocations(ContentProviderUtils contentProvider
         contentProviderUtils.insertTrack(track);
         contentProviderUtils.bulkInsertTrackPoint(trackPoints, track.getId());
     }
+
+    /**
+     * Creates a Waypoint with a photo.
+     *
+     * @param context  The context.
+     * @param trackId  The track id.
+     * @param location The location.
+     * @return the Waypoint created.
+     */
+    public static  Waypoint createWaypointWithPhoto(Context context, long trackId, Location location) {
+        String photoUrl = "";
+        try {
+            File dstFile = new File(FileUtils.getImageUrl(context, trackId));
+            dstFile.createNewFile();
+            Uri photoUri = FileUtils.getUriForFile(context, dstFile);
+            photoUrl = photoUri.toString();
+        } catch (IOException ioe) {
+            throw new RuntimeException(ioe);
```

Comment:
Why not throw the IOException?

Reply:
Because this exception shouldn't never occur or the tests will fail. With RuntimeException I don't need to mark this method as throwing that exception and it results in a cleaner code.

I don't know if it's the best approach for this case.